web/client/components/data/featuregrid/toolbars/Toolbar.jsx

```diff
@@ -163,7 +166,73 @@ const standardButtons = {
         visible={showTimeSyncButton}
         active={timeSync}
         onClick={() => events.setTimeSync && events.setTimeSync(!timeSync)}
-        glyph="time" />)
+        glyph="time" />),
+    snapToFeature: ({snapping, availableSnappingLayers, isSnappingLoading, snappingConfig, mode, mapType, pluginCfg = {}, events = {}}) => (<TSplitButton
+        id="snap-button"
+        keyProp="snap-button"
+        tooltipId={snapping ? "featuregrid.toolbar.disableSnapping" : "featuregrid.toolbar.enableSnapping"}
+        visible={mode === "EDIT" && pluginCfg?.snapTool && mapType === 'openlayers'}
+        onClick={() => {
+            events.toggleSnapping && events.toggleSnapping(!snapping);
+        }}
+        onMount={() => !snappingConfig && events.setSnappingConfigDefaults && events.setSnappingConfigDefaults(pluginCfg?.snapConfig)}
+        onItemClick={(id) => id && events.setSnappingLayer(id)}
+        title={isSnappingLoading ? <Spinner spinnerName="ball-beat" overrideSpinnerClassName="spinner" key="loadingSpinner" noFadeIn /> : <Glyphicon glyph="magnet" />}
+        tooltipPosition="top"
+        className="snap-tool square-button-md no-border"
+        active={!!snapping}
+        pullLeft
+    >
+        {
+            availableSnappingLayers?.length > 1 ?
+                (
+                    <>
+                        <label className="control-label"><Message msgId="featuregrid.toolbar.snapToLayer"/></label>
+                        <div className="layers" style={{"maxHeight": pluginCfg?.snapConfig?.layersListHeight ?? '130px', 'overflowY': 'auto'}}>
+                            { availableSnappingLayers.map(option => <MenuItem active={option.active} onClick={() => !option.active && events.setSnappingLayer(option.value)}>{option.label}</MenuItem>) }
+                        </div>
```

Comment:
Here use a simple "Select" component, so you don't have to fix the height or handle it. 
By default the current layer must be selected

Reply:
Replaced list of items with react-select. As I mentioned previously - it's impossible to render v1 in a portal, so we should think of upgrading this dependency at some point.
Right now dropdown maximum height is capped by feature editor height and react-select options list is capped by the height of dropdown. Scrollbar will appear if either of two cannot fit in feature editor / in dropdown.